Source/Meadow.Contracts/Hardware/Networking/INetworkAdapter.cs

```diff
@@ -39,6 +50,11 @@ public interface INetworkAdapter
     /// </summary>
     event NetworkDisconnectionHandler NetworkDisconnected;
 
+    /// <summary>
+    /// Event raised when a auto-reconnecting to a network has terminaled
+    /// </summary>
+    event NetworkStateHandler ConnectFailed;
```

Comment:
Dont know if we should have it as `NetworkConnectFailed`, since the rest seem to have the `Network` preffix.